snippets/chapel.json

```diff
@@ -22,9 +141,199 @@
+      "config const $1: $2 = $3;"
+    ],
+    "description": "runtime command line argument (./prog --<varname>=<value>)"
+  },
+  "config-param": {
+    "prefix": "configparam",
+    "body": [
+      "config param $1: $2 = $3;"
+    ],
+    "description": "compile-time command line argument (chpl -s<varname>=<value> prog.chpl)"
+  },
+  "module": {
+    "prefix": "module",
+    "body": [
+      "module $1",
+      "  $2",
+      "}"
+    ],
+    "description": "module skeleton"
+  },
+  "use": {
+    "prefix": "use",
+    "body": [
+      "use $1;"
+    ],
+    "description": "module use"
+  },
+  "type": {
+    "prefix": "type",
+    "body": [
+      "type"
+    ],
+    "description": "type declaration"
+  },
+  "bool": {
+    "prefix": "bool",
+    "body": [
+      "bool"
+    ],
+    "description": "boolean type (size implementation dependent)"
+  },
+  "int": {
+    "prefix": "int",
+    "body": [
+      "int"
+    ],
+    "description": "integer type (size=64)"
+  },
+  "uint": {
+    "prefix": "uint",
+    "body": [
+      "uint"
+    ],
+    "description": "unsigned integer type (size=64)"
+  },
+  "real": {
+    "prefix": "real",
+    "body": [
+      "real"
+    ],
+    "description": "real type (size=64)"
+  },
+  "imag": {
+    "prefix": "imag",
+    "body": [
+      "imag"
+    ],
+    "description": "imaginary type (size=64)"
+  },
+  "complex": {
+    "prefix": "complex",
+    "body": [
+      "complex"
+    ],
+    "description": "complex type (size=128)"
+  },
+  "string": {
+    "prefix": "string",
+    "body": [
+      "string"
+    ],
+    "description": "string type"
+  },
+  "record": {
+    "prefix": "record",
+    "body": [
+      "record $1 {",
+      "  $2",
+      "}"
+    ],
+    "description": "record definition"
+  },
+  "class": {
+    "prefix": "class",
+    "body": [
+      "class $1 {",
+      "  $2",
+      "}"
+    ],
+    "description": "class definition"
+  },
+  "method": {
+    "prefix": "method",
+    "body": [
+      "proc $1.$2($3) {",
+      "  $4",
+      "}"
+    ],
+    "description": "define a class method"
+  },
+  "method-override": {
+    "prefix": "override",
+    "body": [
+      "override proc $1.$2($3) {",
+      "  $4",
+      "}"
+    ],
+    "description": "override a class method"
+  },
+  "union": {
+    "prefix": "union",
+    "body": [
+      "union $1 {",
+      "  var $2;",
+      "  var $3;",
+      "}"
+    ],
+    "description": "define a union with two alternatives"
+  },
+  "enum": {
+    "prefix": "enum",
+    "body": [
+      "enum $1 { $2, $3 };"
+    ],
+    "description": "enumeration with two alternatives"
+  },
+  "range": {
+    "prefix": "range",
+    "body": [
+      "$1..$2"
+    ],
+    "description": "bounded range"
+  },
+  "by": {
+    "prefix": "by",
+    "body": [
+      "$1 by $2"
+    ],
+    "description": "strided range"
+  },
+  "counted": {
+    "prefix": "#",
+    "body": [
+      "$1 # $2"
+    ],
+    "description": "counted range"
+  },
```

Comment:
I am not sure just `#` as the prefix is enough to trigger this snippet. I could not get it to work in my testing

Reply:
Weird, I get it to show up (luasnip extension + neovim):

![image](https://github.com/chapel-lang/chapel-vscode/assets/40442285/aaa29eaa-c5f7-491f-a5f5-42654cf68d53)

I can also create a duplicate, with `"prefix": "counted"`

Reply:
It does not auto-show up in VSCode, but if I request snippets with `^Space` it does show up and work. Perhaps a good solution is to do both as you suggest

```
"prefix": ["#", "counted"],
```